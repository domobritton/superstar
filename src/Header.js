import React from 'react';

import superStar from './superstar.svg';

import StarRate from '@material-ui/icons/StarRate';
import styled from 'styled-components';

const Header = () => {
  return (
    <Wrapper>
      <Logo>
        <StarRate />
        <img src={superStar} alt="superstar" />
      </Logo>
    </Wrapper>
  );
};

const Wrapper = styled.header`
  background-color: #282c34;
  height: 80px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

const Logo = styled.div`
  height: 60px;
  svg {
    animation: logo-spin infinite 20s linear;
    font-size: 60px;
    color: gold;
    @keyframes logo-spin {
      from {
        transform: rotate(0deg);
      }
      to {
        transform: rotate(360deg);
      }
    }
  }
  img {
    height: 60px;
  }
`;

export default Header;
