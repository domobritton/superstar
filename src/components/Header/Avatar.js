import React, { Component } from 'react';
import { Avatar } from 'gitstar-components';
import { gql } from 'apollo-boost';
import { Query } from 'react-apollo';

const GET_AVATAR = gql`
  query {
    viewer {
      avatarUrl
    }
  }
`;

export default class UserAvatar extends Component {
  render() {
    if (!localStorage.getItem('token')) {
      return <Avatar {...this.props} />;
    }
    return (
      <Query query={GET_AVATAR}>
        {({ loading, error, data }) => {
          if (loading) return <Avatar {...this.props} />;
          if (error) {
            return <div>:(</div>;
          }

          return <Avatar url={data.viewer.avatarUrl} {...this.props} />;
        }}
      </Query>
    );
  }
}
