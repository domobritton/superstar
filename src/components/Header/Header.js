import React from 'react';
import { STATUS, Avatar } from 'gitstar-components';

import superStar from '../../superstar.svg';

import StarRate from '@material-ui/icons/StarRate';
import styled from 'styled-components';

export const Header = ({ logout, status, CLIENT_ID, REDIRECT_URI }) => {
  return (
    <Wrapper>
      <Logo>
        <StarRate />
        <img src={superStar} alt="superstar" />
      </Logo>
      <Avatar
        style={{
          marginLeft: 'auto',
          transform: `scale(${status === STATUS.AUTHENTICATED ? '1' : '0'})`,
        }}
      />
      {status === STATUS.INITIAL ? (
        <Login
          href={`https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&scope=user&redirect_uri=${REDIRECT_URI}`}
        >
          LOGIN
        </Login>
      ) : (
        <Logout onClick={logout}>LOGOUT</Logout>
      )}
    </Wrapper>
  );
};

const Wrapper = styled.header`
  background-color: gray;
  height: 80px;
  padding: 0 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

const Logo = styled.div`
  height: 60px;
  svg {
    animation: logo-spin infinite 20s linear;
    font-size: 60px;
    color: gold;
    @keyframes logo-spin {
      from {
        transform: rotate(0deg);
      }
      to {
        transform: rotate(360deg);
      }
    }
  }
  img {
    height: 60px;
  }
`;

const Login = styled.a`
  display: inline;
  color: white;
  font-size: 22px;
  &:hover {
    text-decoration-line: none;
    color: gold;
  }
`;

const Logout = styled.button`
  display: inline;
  background: transparent;
  color: white;
  font-size: 22px;
  border: none;
  outline: none;
`;
