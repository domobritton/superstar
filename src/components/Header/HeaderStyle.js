import styled from 'styled-components';

export const Wrapper = styled.header`
  background-color: gray;
  height: 80px;
  padding: 0 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

export const Logo = styled.div`
  height: 60px;
  svg {
    animation: logo-spin infinite 20s linear;
    font-size: 60px;
    color: gold;
    @keyframes logo-spin {
      from {
        transform: rotate(0deg);
      }
      to {
        transform: rotate(360deg);
      }
    }
  }
  img {
    height: 60px;
  }
`;

export const Logout = styled.button`
  display: inline;
  background: transparent;
  color: white;
  font-size: 22px;
  border: none;
  outline: none;
`;

Logout.displayName = 'Button';
