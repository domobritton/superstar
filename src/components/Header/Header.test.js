import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { Header } from './Header';

afterEach(cleanup);

const mockFn = jest.fn();

test('<Header />', () => {
  const { getByTestId, container } = render(<Header logout={mockFn} />);
  const logoutButton = getByTestId('logout');

  expect(container.firstChild).toMatchSnapshot();
  expect(logoutButton.tagName).toBe('BUTTON');
  expect(logoutButton.textContent).toBe('LOGOUT');
  fireEvent.click(logoutButton);
  expect(mockFn).toHaveBeenCalled();
  expect(mockFn).toBeCalledTimes(1);
});
