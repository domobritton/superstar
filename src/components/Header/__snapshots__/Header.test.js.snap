// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> 1`] = `
<header
  class="sc-bdVaJa kGslVD"
>
  <div
    class="sc-bwzfXH kCkLlS"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-1"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 11.3l3.71 2.7-1.42-4.36L15 7h-4.55L9 2.5 7.55 7H3l3.71 2.64L5.29 14z"
        transform="scale(1.33, 1.33)"
      />
      <path
        d="M0 0h18v18H0z"
        fill="none"
        transform="scale(1.33, 1.33)"
      />
    </svg>
    <img
      alt="superstar"
      src="superstar.svg"
    />
  </div>
  <img
    alt="logo"
    class="avatar avatar-small"
    height="32"
    src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
    style="border-radius: 50%; margin-left: auto; transform: scale(0);"
    width="32"
  />
  <button
    class="sc-htpNat gAzCtX"
    data-testid="logout"
  >
    LOGOUT
  </button>
</header>
`;
